Sources/_MatchingEngine/Regex/AST/Conditional.swift

```diff
@@ -0,0 +1,91 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension AST {
+  public struct Conditional: Hashable, _ASTNode {
+    public var location: SourceLocation
+    public var condition: Condition
+
+    public var trueBranch: AST
+    public var pipe: SourceLocation?
```

Comment:
Why optional?

Reply:
The body can be empty, e.g `(?(1))` is valid (albeit pointless)